clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -648,8 +650,17 @@ void CodeGenModule::Release() {
     SPIRVerMD->addOperand(llvm::MDNode::get(Ctx, SPIRVerElts));
     // We are trying to look like OpenCL C++ for SPIR-V translator.
     // 4 - OpenCL_CPP, 100000 - OpenCL C++ version 1.0
+    // 6 - ESIMD, if any kernel or function is an explicit SIMD one
+    int Lang = 4;
+
+    if (std::find_if(TheModule.begin(), TheModule.end(),
```

Comment:
llvm::find_if takes a range!  Also, you can use 'auto' for the param, and don't need the trailing return type in C++14 (of which clang is now).

Finally, I think this could be llvm::any_of:

    int Lang = llvm::any_of(TheModule, [](const auto &F) { return F.getMetadata("sycl_explicit_simd"); }) ? 6 : 4;

Reply:
ok